persistence/shared_sql.go

```diff
@@ -1,20 +1,21 @@
 package persistence
 
 import (
+	"context"
 	"encoding/hex"
 	"fmt"
-
 	"github.com/jackc/pgx/v4"
+
 	"github.com/pokt-network/pocket/persistence/schema"
 	"github.com/pokt-network/pocket/shared/types"
 )
 
 // IMPROVE(team): Move this into a proto enum
 const (
-	UndefinedStakingStatus = iota
-	UnstakedStatus
-	UnstakingStatus
-	StakedStatus
+	UndefinedStakingStatus = 3
```

Comment:
```
	UnstakedStatus = iota
	UnstakingStatus 
	StakedStatus
```

This would result in a unique but non-deterministic number set so this won't work.

Fine to change the order though

Reply:
Order changed with [d2edd36](https://github.com/pokt-network/pocket/pull/140/commits/d2edd36ddba729097db8ddbe6b82a26a3f64c4f4)

Reply:
Note that this isn't a blocker for the PR but I want to understand why you said it's a `uniqute but non-deterministic number`.

My understanding of `iota` in Golang is that it is an enumeration that increments by 1 based on order per the screenshot and reference below.


![Screen Shot 2022-08-11 at 1 19 06 PM](https://user-images.githubusercontent.com/1892194/184194538-e90ca7af-531b-4351-b8f0-b6a6025e6ff9.png)
Reference: https://www.gopherguides.com/articles/how-to-use-iota-in-golang


I ran `make test_all` after replacing your updated code:

```go
const (
	UndefinedStakingStatus = 0
	UnstakingStatus        = 1
	StakedStatus           = 2
	UnstakedStatus         = 3
)
```

with:

```go
const (
	UndefinedStakingStatus = iota
	UnstakingStatus
	StakedStatus
	UnstakedStatus
)
```
and everything passed.

Wanted to confirm this is okay before I push or if I'm missing something?

Reply:
> a unique but non-deterministic number

Admittedly, I wasn't aware that iota always started with zero and incremented by one.
*I thought it was guaranteed unique, but not always starting at zero* which appears to be incorrect.

With that being said, let me state my case about why I don't think Iota's are a good choice for this:
- There are only 3 or 4 possible status' so it's not a huge verbosity issue
- The syntax sugar of the iota is not worth the language specific rules that come along with the change, example:
  - Code order of constants matters
  - Iotas start with zero or iota+1 starts with 1 
  - Careful with creating two iotas in the same const() block
<img width="543" alt="Screen Shot 2022-08-11 at 4 42 37 PM" src="https://user-images.githubusercontent.com/18633790/184238664-233717f5-f2e2-4134-a3ee-82b1b721ffb3.png">

vs

<img width="421" alt="Screen Shot 2022-08-11 at 4 47 05 PM" src="https://user-images.githubusercontent.com/18633790/184239299-816cf899-ca23-4b98-8a0f-6ef3b16e669f.png">


- **TL;DR** I still like being explicit considering how a change like this affects the app hash and other language implementations will need to match these constant values exactly.

Reply:
@Olshansk noticed you still made the change to iota. Lmk if you want to take it offline

Reply:
Reversed it back.